core/state_processor_rip7560.go

```diff
@@ -18,17 +16,50 @@ import (
 	"strings"
 )
 
-const MaxContextSize = 65536
+const MAGIC_VALUE_SENDER = uint64(0xbf45c166)
+const MAGIC_VALUE_PAYMASTER = uint64(0xe0e6183a)
+const MAGIC_VALUE_SIGFAIL = uint64(0x31665494)
+const PAYMASTER_MAX_CONTEXT_SIZE = 65536
 
-var MAGIC_VALUE_SENDER = [20]byte{0xbf, 0x45, 0xc1, 0x66}
-var MAGIC_VALUE_PAYMASTER = [20]byte{0xe0, 0xe6, 0x18, 0x3a}
-var MAGIC_VALUE_SIGFAIL = [20]byte{0x31, 0x66, 0x54, 0x94}
+func PackValidationData(authorizerMagic uint64, validUntil, validAfter uint64) []byte {
+	t := new(big.Int).SetUint64(uint64(validAfter))
+	t = t.Lsh(t, 48).Add(t, new(big.Int).SetUint64(validUntil&0xffffff))
+	t = t.Lsh(t, 160).Add(t, new(big.Int).SetUint64(uint64(authorizerMagic)))
+	return common.LeftPadBytes(t.Bytes(), 32)
+}
+
+func UnpackValidationData(validationData []byte) (uint64, uint64, uint64) {
+	authorizerMagic := new(big.Int).SetBytes(validationData[:20]).Uint64()
+	validUntil := new(big.Int).SetBytes(validationData[20:26]).Uint64()
+	validAfter := new(big.Int).SetBytes(validationData[26:32]).Uint64()
+	return authorizerMagic, validUntil, validAfter
+}
+
+func UnpackPaymasterValidationReturn(paymasterValidationReturn []byte) ([]byte, []byte, error) {
```

Comment:
It looks like the return type of the validatePaymasterTransaction function is different from what we're used to. Do we need to modify the paymaster contract end?

Reply:
Imo this is quite an open question, since this is the way of ERC-4337. For better compatibility, I think this should be discussed more.